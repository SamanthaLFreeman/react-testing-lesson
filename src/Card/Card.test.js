import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Card
      title="Blah"
      description="Blah blah blah"
      id={1}
      removeIdea={jest.fn()}
      />);

      expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if the card is favorited', () => {
    const wrapper = shallow(<Card
      title="Blah"
      description="Blah blah blah"
      id={1}
      removeIdea={jest.fn()}
      isFavorite={true}
    />);

    expect(wrapper).toMatchSnapshot()
  })

  it('should call the removeIdea prop with the card\'s id when clicked', () => {
    //Setup
    const removeIdeaMock = jest.fn()
    const wrapper = shallow(<Card
      title="Blah"
      description="Blah blah blah"
      id={1}
      removeIdea={removeIdeaMock}
      isFavorite={true}
    />);
    
    //Execution
    wrapper.find('button').simulate('click');

    //Expectation
    expect(removeIdeaMock).toHaveBeenCalledWith(1);
  })
})
