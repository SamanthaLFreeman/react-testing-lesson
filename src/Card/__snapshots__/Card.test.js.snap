// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if the card is favorited 1`] = `
<section
  className="favorite"
>
  <h3>
    Blah
  </h3>
  <p>
    Blah blah blah
  </p>
  <button
    onClick={[Function]}
  >
    ðŸ—‘
  </button>
</section>
`;

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<section
  className="card"
>
  <h3>
    Blah
  </h3>
  <p>
    Blah blah blah
  </p>
  <button
    onClick={[Function]}
  >
    ðŸ—‘
  </button>
</section>
`;
